src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/Microsoft.NET.Sdk.Razor.SourceGenerators.Test.csproj

```diff
@@ -34,19 +34,29 @@
     <ProjectReference Include="..\Microsoft.AspNetCore.Razor.Test.Common\Microsoft.AspNetCore.Razor.Test.Common.Compiler.csproj" />
   </ItemGroup>
 
-  <!--
-    Include ASP.NET Core DLLs.
-
-    Even though we reference `win-x64` package, it doesn't matter, we don't use any native DLLs.
-    It's just to work around the fact that ASP.NET Core DLLs are not normally published to NuGet.
-    And we don't want to reference ASP.NET Core SDK as that would be tied to our target SDK version.
-  -->
   <PropertyGroup>
     <ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App.Runtime.win-x64" ExcludeAssets="all" GeneratePathProperty="true" />
-    <Reference Include="$(PkgMicrosoft_AspNetCore_App_Runtime_win-x64)\runtimes\win-x64\lib\net7.0\*.dll" />
+    <PackageReference Include="Microsoft.AspNetCore.App.Runtime.$(NetCoreSDKRuntimeIdentifier)" ExcludeAssets="all" GeneratePathProperty="true" />
   </ItemGroup>
 
+  <Choose>
+    <When Condition="$(NetCoreSDKRuntimeIdentifier) == 'win-x64'">
+      <ItemGroup>
+        <Reference Include="$(PkgMicrosoft_AspNetCore_App_Runtime_win-x64)\runtimes\win-x64\lib\net7.0\*.dll" />
```

Comment:
I wonder if MSBuild supports dynamic properties so you could avoid the `Choose`, but probably not.

```csproj
<Reference Include="$(PkgMicrosoft_AspNetCore_App_Runtime_$(NetCoreSDKRuntimeIdentifier))\runtimes\$(NetCoreSDKRuntimeIdentifier)\lib\net7.0\*.dll" />
```

Reply:
I tried a dozen variations on this unfortunately :cry:

Reply:
So, this *is* possible if you abuse item groups but let's not do that. For reference:

<details>

<summary>MSBuild abuse</summary>

```xml
<Project>

	<PropertyGroup>
		<PkgMicrosoft_AspNetCore_App_Runtime_win-x64>c:\win\specific\path</PkgMicrosoft_AspNetCore_App_Runtime_win-x64>
		<PkgMicrosoft_AspNetCore_App_Runtime_linux-x64>c:\linux\specific\path</PkgMicrosoft_AspNetCore_App_Runtime_linux-x64>
		<NetCoreSDKRuntimeIdentifier>win-x64</NetCoreSDKRuntimeIdentifier>		
	</PropertyGroup>
	
	<ItemGroup>
		<Expander Include="PkgMicrosoft_AspNetCore_App_Runtime_$(NetCoreSDKRuntimeIdentifier)">
			<PackagePath>$(%(Identity))</PackagePath>
		</Expander>
	</ItemGroup>
	
	<PropertyGroup>
		<PackagePath>@(Expander->Metadata('PackagePath'))</PackagePath>
	</PropertyGroup>
	
	
	<Target Name="build">
                <!-- prints c:\win\specific\path -->
		<Message Importance="high" Text="$(PackagePath)" />
	</Target>

</Project>
```

</details>

Reply:
Note to self: Chris knows some cursed MSBuild things.